src/tools/tests/recipe2plan-test.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {Manifest} from '../../runtime/manifest.js';
+import {StorageKeyRecipeResolver} from '../recipe2plan.js';
+import {assert} from '../../platform/chai-node.js';
+
+describe('recipe2plan', () => {
+  it('Long + Long: If ReadingRecipe is long running, it is a valid use case', async () => {
+    const manifest = await Manifest.parse(`\
+    particle Reader
+      data: reads Thing {name: Text}
+
+    // TODO(alxr): Resolve these types later
+    // particle Writer
+    //   data: writes Product Thing {name: Text, price: Number}
+
+    particle Writer
+       data: writes Thing {name: Text}
+    
+    @trigger
+      launch startup
+      arcId myArcId
+    recipe WritingRecipe
+      thing: create persistent 'my-handle-id' 
+      Writer
+        data: writes thing
+
+    @trigger
+      launch startup
+      arcId otherArcId
+    recipe ReadingRecipe
+      data: map 'my-handle-id'
+      Reader
+        data: reads data`);
+
+    const resolver = new StorageKeyRecipeResolver(manifest);
+    // @ts-ignore
```

Comment:
What's this for?

Reply:
I just addressed it -- now it's unnecessary: I just changed the return of `resolve` to be an async generator instead of iterator. The latter cause typing issues.